package/MDAnalysis/analysis/distances.py

```diff
@@ -55,6 +54,9 @@ def contact_matrix(coord, cutoff=15.0, returntype="numpy", box=None):
     .. versionchanged:: 0.11.0
        Keyword *suppress_progmet* and *progress_meter_freq* were removed.
     '''
+    
+    from scipy import sparse      
```

Comment:
Let's do the following to get the PR done (because @jandom has already spend a lot of effort on it and this part is not his fault really and technically #666 is not active yet): leave as is for this PR but raise an issue for consistent handling of Scipy  sparse matrices in the module, namely a way to do a guarded import at the top and raise an appropriate error in the function if Scipy.sparse is not available. 

--
Oliver Beckstein
email: orbeckst@gmail.com


Am Feb 23, 2016 um 3:25 schrieb Richard Gowers <notifications@github.com>:

> In package/MDAnalysis/analysis/distances.py:
> 
> > @@ -55,6 +54,9 @@ def contact_matrix(coord, cutoff=15.0, returntype="numpy", box=None):
> >      .. versionchanged:: 0.11.0
> >         Keyword *suppress_progmet* and *progress_meter_freq* were removed.
> >      '''
> > +    
> > +    from scipy import sparse      
> It doesn't really make sense here though. If we put the import at the top, then the entire module is inaccessible to non scipy people, whilst it looks like there's only one option in one function that needs it.
> 
> —
> Reply to this email directly or view it on GitHub.
>